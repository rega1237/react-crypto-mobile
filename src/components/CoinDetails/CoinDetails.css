.presentation {
  display: flex;
  flex-direction: column;
  background-color: var(--pink);
}

.back-icon {
  width: 1rem;
  margin: 1rem 0.5rem;
}

.img-name {
  display: flex;
  gap: 1.5rem;
  padding: 1rem;
}

.img-coin-details {
  width: 15%;
 }

.name-rank h1 {
  font-family: var(--gill);
  color: var(--white);
}

.name-rank h2 {
  font-family: var(--lato);
  color: var(--white);
}

.description-element {
  list-style: none;
  padding: 1.2rem;
  color: var(--white);
}

.description-element:nth-child(even) {
  background-color: var(--pink);
}

.description-element h2 {
  font-family: var(--gill);
}

.description-element h3 {
  font-family: var(--lato);
  text-align: end;
}

.error {
  text-align: center;
  font-family: var(--gill);
  color: var(--white);
  padding: 1rem;
  margin-top: 3rem;
}
