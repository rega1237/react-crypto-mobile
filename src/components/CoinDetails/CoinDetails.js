import { NavLink, useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import back from '../../icon/back.png';
import './CoinDetails.css';
import Header from '../Header/Header';

const CoinDetails = () => {
  const { id } = useParams();
  const coinStore = useSelector((state) => state.cryptoReducer);
  const selectCoin = coinStore.filter((element) => element.id === id);
  const coin = selectCoin[0];

  if (coin !== undefined) {
    return (
      <div className="details-container">
        <div className="presentation">
          <NavLink to="/">
            <img className="back-icon" src={back} alt="" />
          </NavLink>
          <Header />
          <div className="img-name">
            <img className="img-coin-details" src={coin.img} alt={`${coin.name} logo`} />
            <div className="name-rank">
              <h1>{`${coin.name} (${coin.symbol})`}</h1>
              <h2>{`Rank #${coin.rank}`}</h2>
            </div>
          </div>
        </div>

        <ul>
          <li className="description-element">
            <h2>Price:</h2>
            <h3>{`${coin.priceUsd.toFixed(3)} USD`}</h3>
          </li>
          <li className="description-element">
            <h2>Market cap:</h2>
            <h3>{`${coin.marketCapUsd.toFixed(3)} USD`}</h3>
          </li>
          <li className="description-element">
            <h2>Supply:</h2>
            <h3>{`${coin.supply}`}</h3>
          </li>
          <li className="description-element">
            <h2>Max Supply:</h2>
            <h3>{`${coin.maxSupply}`}</h3>
          </li>
          <li className="description-element">
            <h2>Volume last 24 hr:</h2>
            <h3>{`${coin.volumeUsd24Hr.toFixed(3)} USD`}</h3>
          </li>
        </ul>
      </div>
    );
  }
  return (
    <div className="details-container">
      <div className="presentation">
        <NavLink to="/">
          <img className="back-icon" src={back} alt="" />
        </NavLink>
        <Header />
      </div>
      <div className="error">
        <h1>Oops! nothing To show here</h1>
        <h2>Please go back and select the Currency you want to check!</h2>
      </div>
    </div>
  );
};

export default CoinDetails;
