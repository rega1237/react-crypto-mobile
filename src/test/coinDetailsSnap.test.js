import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import CoinDetails from '../components/CoinDetails/CoinDetails';

test('selected snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <CoinDetails />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
