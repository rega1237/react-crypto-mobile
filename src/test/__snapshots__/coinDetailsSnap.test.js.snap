// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selected snapshot 1`] = `
<div
  className="details-container"
>
  <div
    className="presentation"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt=""
        className="back-icon"
        src="back.png"
      />
    </a>
    <header>
      <img
        alt="company logo"
        src="logo.png"
      />
    </header>
  </div>
  <div
    className="error"
  >
    <h1>
      Oops! nothing To show here
    </h1>
    <h2>
      Please go back and select the Currency you want to check!
    </h2>
  </div>
</div>
`;
